src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs

```diff
@@ -184,85 +199,102 @@ public static GridInvokeMethodResult ForwardCopyTo(GridFunctionParameter paramet
+                {
+                    targetUserId = user.Ident;
+                }
+                else
+                    workflowOrganzisationId = (Guid?)item.Id;
 
-                var workflowOperation = new WorkflowOperation
+                var comment = new Framework.Extension.InstanceDataCommentModel
                 {
-                    DocumentId = documentId,
-                    DocumentPath = documentPathId,
+                    Comment = commentText,
+                    UserGroupVisibility = Visibility.Hidden,
                     UserId = sessionService.CurrentSession.UserId,
-                    TargetUserId = targetUserId,
-                    CreateDateTime = DateTime.Now,
-                    UpdateDateTime = DateTime.Now,
-                    ActionName = "forward",
-                    WorkflowId = workflowId,
-                    Guid = Guid.NewGuid()
+                    InstanceDataGuid = Guid.NewGuid(),
+                    StackGuid = Guid.Parse("12c9b95b-bd33-4fa0-9ca1-05e11122018c") // TODO: Remove magic string
                 };
-                if (itemBox.GetSelectedItemCell("InternalType").ToString() == "Group")
-                {
-                    workflowOperation.OperationType = WorkflowOperationType.WorkflowOrganizationUnit; 
-                    workflowOperation.WorkflowOrganizationId = workflowOrganizationId; 
-                }
 
-                try
+                foreach (var row in parameter.GetSelectedRowsAsDataRow())
                 {
-                    workflowOperationService.ForwardCopyTo(workflowOperation);
-                }
-                catch (DocumentWorkflowException ex)
-                {
-                    Log.LogManagerInstance.Instance.Error("Could not forward document in workflow", ex);
+                    var documentId = (Guid)row["Guid"];
+                    var documentPathId = (Guid)row["DocumentPathId"];
+                    // The grid needs a the column workflow id 
+                    var workflowId = (Guid)fileStructureDocumentPathService.Get(documentPathId).WorkflowId;
 
-                    // TODO: Add localization
-                    MessageBox.Show("Workflow für Zielbenutzer nicht gefunden.", "Workflow nicht gefunden", MessageBoxButton.OK, MessageBoxImage.Information);
-                    return GridInvokeMethodResult.NoGridRefresh();
-                }
+                    var workflowOperation = new WorkflowOperation
+                    {
+                        DocumentId = documentId,
+                        DocumentPath = documentPathId,
+                        UserId = sessionService.CurrentSession.UserId,
+                        TargetUserId = targetUserId,
+                        CreateDateTime = DateTime.Now,
+                        UpdateDateTime = DateTime.Now,
+                        ActionName = "forward",
+                        WorkflowId = workflowId,
+                        Guid = Guid.NewGuid(),
+                    };
```

Comment:
a lot of this seems to be implemented twice, is it worth to move it to a seperate method?